src/ui/components/pages/NewWalletName.tsx

```diff
@@ -1,174 +1,148 @@
-import * as styles from './styles/newaccountname.styl';
+import * as styles from 'ui/components/pages/newWalletName.module.css';
 import * as React from 'react';
-import { connect } from 'react-redux';
-import { Trans } from 'react-i18next';
-import { addUser, newAccountName, setUiState } from '../../actions';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  createAccount,
+  newAccountName,
+  selectAccount,
+  setTab as resetTab,
+} from '../../actions';
 import { Button, Error, Input } from '../ui';
 import { CONFIG } from '../../appConfig';
 import { WalletTypes } from '../../services/Background';
-import { AppState } from 'ui/store';
-
-class NewWalletNameComponent extends React.Component {
-  readonly props;
-  readonly state = { disabled: false, errors: [] } as any;
-
-  passwordError: boolean;
-
-  constructor(params) {
-    super(params);
-    this.props.setUiState({
-      account: null,
-    });
-  }
-
-  static validateName(name: string, accounts) {
-    const errors = [];
-    const names = accounts.map(({ name }) => name);
-
-    if (name.length < CONFIG.NAME_MIN_LENGTH) {
-      errors.push({
-        code: 1,
-        key: 'newAccountName.errorRequired',
-        msg: 'Required name',
-      });
-    }
+import { useAppDispatch, useAppSelector } from 'accounts/store';
+import { PAGES } from 'ui/pageConfig';
+
+export function NewWalletName({ setTab }) {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const account = useAppSelector(state => state.localState.newAccount);
+  const accounts = useAppSelector(state => state.accounts);
+  const [accountName, setAccountName] = React.useState<string>('');
+  const [pending, setPending] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<string>('');
+  const [touched, setTouched] = React.useState<boolean>(false);
+
+  const existedAccount = accounts.find(
```

Comment:
I guess it's not needed to search for an existing account here, because switching is done on previous step. Or is it needed in some cases?

Reply:
This prevents some bugs with multiple accounts tabs.